microsoft/testsuites/dpdk/dpdktestpmd.py

```diff
@@ -111,45 +112,69 @@ def _install(self) -> bool:
         self.dpdk_path = self.node.working_path.joinpath(self._dpdk_repo_path_name)
         if result.exit_code == 0:  # tools are already installed
             return True
-        if self._install_dependencies():
-            node = self.node
-            git_tool = node.tools[Git]
-            echo_tool = node.tools[Echo]
+        self._install_dependencies()
+        node = self.node
+        git_tool = node.tools[Git]
+        echo_tool = node.tools[Echo]
+
+        if self._dpdk_source and self._dpdk_source.endswith(".tar.gz"):
+            wget_tool = node.tools[Wget]
+            tar_tool = node.tools[Tar]
+            if self._dpdk_branch:
+                node.log.warn(
+                    (
+                        "DPDK tarball source does not need dpdk_branch defined. "
+                        "User-defined variable dpdk_branch will be ignored."
+                    )
+                )
+            working_path = node.working_path.as_posix()
+            wget_tool.get(
+                self._dpdk_source,
+                working_path,
+            )
+            dpdk_filename = self._dpdk_source.split("/")[-1]
+            node.log.info(node.execute("ls -la", cwd=node.working_path))
+            dpdk_foldername = tar_tool.extract(
+                node.working_path.joinpath(dpdk_filename).as_posix(),
```

Comment:
the as_posix is not necessary. The working_path itself is already posix, if the remote node is.